src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -108,6 +108,12 @@ BlockIO InterpreterCreateUserQuery::execute()
         throw Exception(ErrorCodes::BAD_ARGUMENTS,
             "Authentication type NO_PASSWORD must be explicitly specified, check the setting allow_implicit_no_password in the server configuration");
 
+    if (!query.attach && query.temporary_password_for_checks)
```

Comment:
So we won't check password complexity for existing users? This should be backward compatible then

Reply:
Yes, these will only be applied to the new passwords